core/src/gui/widgets/waterfall.cpp

```diff
@@ -779,7 +777,7 @@ namespace ImGui {
             float dataRange = waterfallMax - waterfallMin;
             for (int j = 0; j < dataWidth; j++) {
                 pixel = (std::clamp<float>(latestFFT[j], waterfallMin, waterfallMax) - waterfallMin) / dataRange;
-                int id = (int)(pixel * (WATERFALL_RESOLUTION - 1));
+                int id = std::clamp<int>((int)(pixel * (WATERFALL_RESOLUTION - 1)), 0, WATERFALL_RESOLUTION);
```

Comment:
Value is already clamped on line 779.

Reply:
I rewrote this place and added comment to explain why this is needed.

Reply:
I can't accept those changes. The values here should NEVER be NAN. The only case it could be NAN is if your source module is giving bad data. Checking for NAN here decrease performance.

Reply:
Ok, I understood. I was thinking that this can be useful if source produce bad data by some reason (which happened for me sometimes, so now I will try to figure why such data produced and try to solve this in source).